lib/src/signed_video_h26x_auth.c

```diff
@@ -353,6 +459,79 @@ set_validation_status_of_items_used_in_gop_hash(h26x_nalu_list_t *nalu_list, cha
   return num_marked_items;
 }
 
+/**
+ * Verifies the integrity of the GOP hash in the video, ensuring that the data
+ * within the GOP is authentic and complete. Updates the expected and received
+ * NALU counts, and returns true if the verification is successful.
+ */
+static bool
+verify_hashes_with_sei(signed_video_t *self, int *num_expected_nalus, int *num_received_nalus)
+{
+  assert(self);
+
+  int num_expected_hashes = -1;
+  int num_received_hashes = -1;
+  char validation_status = 'P';
+  h26x_nalu_list_t *nalu_list = self->nalu_list;
+  h26x_nalu_list_item_t *sei = h26x_nalu_list_get_next_sei_item(nalu_list);
+  svrc_t status = SV_UNKNOWN_FAILURE;
+
+  // Mark all the NALUs that belong to the GOP to verify.
+  add_hashes_for_computing_gop_hash(self, sei);
```

Comment:
Wouldn't it be more intuitive to put this preparation operation in `prepare_for_validation(...)`?
And at the same time compute the gop hash. Similar to the existing `compute_gop_hash(...)`. Then we keep the same principle and avoid any potential mismatches between which NALUs that are selected in `prepare_for_validation()` and here.